enhancements/ingress/gateway-api-crd-life-cycle-management.md

```diff
@@ -0,0 +1,496 @@
+>   in the kubelet? How does an n-2 kubelet without this feature available behave
+>   when this feature is used?
+> - Will any other components on the node change? For example, changes to CSI, CRI
+>   or CNI may require updating that component before the kubelet.
+
+_TBD: Do we describe version skew with layered products here?_
+
+## Operational Aspects of API Extensions
+
+> Describe the impact of API extensions (mentioned in the proposal section, i.e. CRDs,
+> admission and conversion webhooks, aggregated API servers, finalizers) here in detail,
+> especially how they impact the OCP system architecture and operational aspects.
+> 
+> - For conversion/admission webhooks and aggregated apiservers: what are the SLIs (Service Level
+>   Indicators) an administrator or support can use to determine the health of the API extensions
+> 
+>   Examples (metrics, alerts, operator conditions)
+>   - authentication-operator condition `APIServerDegraded=False`
+>   - authentication-operator condition `APIServerAvailable=True`
+>   - openshift-authentication/oauth-apiserver deployment and pods health
+> 
+> - What impact do these API extensions have on existing SLIs (e.g. scalability, API throughput,
+>   API availability)
+> 
+>   Examples:
+>   - Adds 1s to every pod update in the system, slowing down pod scheduling by 5s on average.
+>   - Fails creation of ConfigMap in the system when the webhook is not available.
+>   - Adds a dependency on the SDN service network for all resources, risking API availability in case
+>     of SDN issues.
+>   - Expected use-cases require less than 1000 instances of the CRD, not impacting
+>     general API throughput.
+> 
+> - How is the impact on existing SLIs to be measured and when (e.g. every release by QE, or
+>   automatically in CI) and by whom (e.g. perf team; name the responsible person and let them review
+>   this enhancement)
+> 
+> - Describe the possible failure modes of the API extensions.
+> - Describe how a failure or behaviour of the extension will impact the overall cluster health
+>   (e.g. which kube-controller-manager functionality will stop working), especially regarding
+>   stability, availability, performance and security.
+> - Describe which OCP teams are likely to be called upon in case of escalation with one of the failure modes
+>   and add them as reviewers to this enhancement.
+
+_TBD: Do we need to describe anything here?_
+
+## Support Procedures
+
+### Conflicting CRDs
+
+If the Ingress Operator detects the presence of a conflicting version of the
+Gateway API CRDs, it updates the ingress clusteroperator to report a `Degraded`
+status condition with status `True` and a message explaining the situation:
+
+_TBD: Insert example output from `oc get clusteroperators/ingress -o yaml`._
+
+In this situation, the cluster-admin is expected to verify that workload would
+not be broken by handing life-cycle management of the CRDs over to the Ingress
+Operator:
```

Comment:
If they cannot verify this, or I guess, they verify it and determine their workload would be broken, we need to explain that path too

Reply:
Right, we will probably need to describe how workload could be broken (e.g. third-party controller fails to start because it needs newer CRDs), how to predict or detect it, and how to handle it (even if it comes down to, "Work with Red Hat and the third-party vendor to align on a common CRD version.").